src/vsr/replica.zig

```diff
@@ -5992,15 +6308,12 @@ pub fn ReplicaType(
         fn view_jump(self: *Self, header: *const Header) void {
             const to: Status = switch (header.command) {
                 .prepare, .commit => .normal,
-                .start_view_change, .do_view_change, .start_view => .view_change,
-                else => unreachable,
+                // When we are recovering_head we can't participate in a view-change anyway.
+                // But there is a chance that the primary is actually running, despite the DVC/SVC.
+                .do_view_change, .start_view_change => if (self.status == .recovering_head) Status.normal else .view_change,
```

Comment:
Maybe spell `.start_view => return` explicitly with a comment that that case we handle manually in `on_start_view`?